near-sdk/src/environment/env.rs

```diff
@@ -708,3 +712,131 @@ pub fn state_write<T: borsh::BorshSerialize>(state: &T) {
     let data = state.try_to_vec().expect("Cannot serialize the contract state.");
     storage_write(STATE_KEY, &data);
 }
+
+/// Returns `true` if the contract state exists and `false` otherwise.
+pub fn state_exists() -> bool {
+    storage_has_key(STATE_KEY)
+}
+
+// ##################
+// # Helper methods #
+// ##################
+
+/// Returns `true` if the given account ID is valid and `false` otherwise.
+pub fn is_valid_account_id(account_id: &[u8]) -> bool {
```

Comment:
Ideally, this logic should live in `near-vm-logic` or `primitives` and we should be importing it here.

Reply:
Yep, good point. I also wanted to validate account ID in the `near-vm-logic`, so we can throw errors earlier.